### ISSUE TITLE
docker collect_container_size option causes EBS IO saturation

### ISSUE THREAD

**Issue by alexmac** [DataDog/datadog-agent#1052]
**Describe what happened:**

enabling `collect_container_size: true` in `/etc/dd-agent/conf.d/docker_daemon.yaml` causes constant high IO to the block device being used by docker (AWS EBS GP2 volume used by an Amazon Linux machine part of an ECS cluster).

Once the IO credits are exhausted docker is unable to function properly.

The docker logs are full of messages like this: 
```
time="2018-01-14T00:15:59.266286595Z" level=error msg="Failed to compute size of container rootfs dcb786130fb98f20a75152f7f0069957b5409fb632e3fc6cd36d167856f0d426: mount does not exist" 
```

**Describe what you expected:**

Either more of an aggressive warning in the docs suggesting that enabling this option is a terrible idea on AWS EBS volumes, or to have the option removed

**Steps to reproduce the issue:**

I've not got a solid repro, other than spin up an EC2 instance using an EBS volume (200Gb GP2 in my case) then deploy a dozen large 1-2Gb containers to it and watch iotop as the root docker process spends 99% of its time performing IO

**Comment by xvello** [comment]
Hi @alexmac as you mention the `/etc/dd-agent/conf.d/docker_daemon.yaml` file, could you confirm to me whether you are using the agent5 (`datadog/docker-dd-agent` image) or the agent6 (`datadog/agent` image)?

This is a known issue with docker and why the option is disabled by default. Unfortunately, as of today, the daemon does not expose another way to get this information.

**Comment by alexmac** [comment]
I'm running whatever this script installs: `https://raw.githubusercontent.com/DataDog/dd-agent/master/packaging/datadog-agent/source/install_agent.sh`

"This is a known issue with docker and why the option is disabled by default" - yeah, there's a warning that says you shouldn't enable it if the docker ps -a command runs slowly, however that runs just fine for me - I think minimally you should update the warning to make it clear it's a potentially fatal mistake that will brick your EC2 instance if enabled :)

**Comment by brendanlong** [comment]
We just ran into this in ECS in both Agent v5 and v6: https://github.com/aws/amazon-ecs-agent/issues/1489#issuecomment-409677161

I also saw the note in the comment, ran `docker ps -a`, confirmed that it was very fast, and at first thought it wasn't the problem (a lot of changes happened at the same time so this took me a while to debug). It would be nice if the comment specifically mentioned that if you see problems with docker.sock being slow or broken, that this config is a known cause of that.

**Comment by xvello** [comment]
Hello,

https://github.com/DataDog/datadog-agent/pull/1915, shipped in 6.4.1 restores the `collect_container_size_frequency` option to reduce the collection frequency, and sets the default value to 5 (collect every 75 seconds instead of 15).
